src/lightning/fabric/strategies/fsdp.py

```diff
@@ -821,9 +827,8 @@ def _load_raw_module_state(path_or_ckpt: Union[Path, Dict[str, Any]], module: Mo
 
     from torch.distributed.fsdp import FullyShardedDataParallel as FSDP
 
-    # This is inefficient, as multiple copies of the checkpoint are held in CPU memory at once.
-    # There is currently no other way because `summon_full_params` does not support write-back from rank 0 only.
-    state_dict = torch.load(path_or_ckpt, map_location="cpu") if not isinstance(path_or_ckpt, dict) else path_or_ckpt
+    # Use `lazy_load` instead of `torch.load` here to avoid storing a copy of the full checkpoint per rank
+    state_dict = _lazy_load(path_or_ckpt) if isinstance(path_or_ckpt, Path) else path_or_ckpt
```

Comment:
Where do we pass `path_or_ckpt` as a path? I hadn't noticed it before but I find it strange to allow both options.

Reply:
Happy to refactor this in a follow-up. The logic has quickly grown over the last few weeks and we need to take care of it, I agree.

Two places where we call it:
https://github.com/Lightning-AI/lightning/blob/6511ac28759718a524dd00e627c186fb6baea763/src/lightning/fabric/strategies/fsdp.py#L510-L512

https://github.com/Lightning-AI/lightning/blob/6511ac28759718a524dd00e627c186fb6baea763/src/lightning/fabric/strategies/fsdp.py#L575-L577

Reply:
Sounds good to me